tests/compile-fail/never_say_never.rs

```diff
@@ -4,7 +4,7 @@
 fn main() {
     let y = &5;
     let x: ! = unsafe {
-        *(y as *const _ as *const !) //~ ERROR entered unreachable code
+        *(y as *const _ as *const !) //~ ERROR tried to access a dead local variable
```

Comment:
See https://github.com/rust-lang/rust/commit/84d1f6aa82123c2951042aff99e43bdb894d3d81 - this is the wrong microoptimizations IMO, one could just emit `unreachable` instead of creating the temporary at all.